soroban-sdk/src/testutils.rs

```diff
@@ -438,6 +438,12 @@ pub trait Address {
     fn generate(env: &Env) -> crate::Address;
 }
 
+pub trait MuxedAddress {
+    /// Create a new MuxedAddress from a provided account public key and
+    /// multiplexing identifier.
+    fn from_account_id(env: &Env, account_key: &[u8; 32], id: u64) -> crate::MuxedAddress;
```

Comment:
I think the account key input should be an Address. Users don't usually pass account keys around like this. The naming is also ambiguous to if it is the public key or secret key.
```suggestion
    fn new(env: &Env, address: &Address, muxed_id: u64) -> crate::MuxedAddress;
```

Reply:
That's the most obvious option, but it would require invalid input handling (which is confusing for the user as not every address will be accepted), and (more annoyingly) this would require a generator for the account addresses, which opens a whole new can of worms. I'd like to avoid explicitly adding account address generator for tests for the time being. I've tried a `generate` interface, as well as function for creating addresses with a different mux id. This way account addresses don't need to 'leak' into the usual address test utils.

Reply:
Folks can use the existing `Address::generate`?

Reply:
No, the existing `Address::generate` creates contract addresses only and we can't simply switch it to be account address, as that would break SAC interactions. I don't think it's worth the effort to try and support that, given that what we're trying to do here is only necessary for a narrow use case (custom token implementations) that doesn't need to involve SAC.